compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -895,9 +895,16 @@ private boolean isAssignable(BType source, BType target, Set<TypePair> unresolve
             return isFiniteTypeAssignable((BFiniteType) source, target, unresolvedTypes);
         }
 
-        if ((targetTag == TypeTags.UNION || sourceTag == TypeTags.UNION) &&
-                isAssignableToUnionType(source, target, unresolvedTypes)) {
-            return true;
+        if ((targetTag == TypeTags.UNION || sourceTag == TypeTags.UNION)) {
+            boolean isAssignableToUnion = isAssignableToUnionType(source, target, unresolvedTypes);
+            if (isAssignableToUnion) {
```

Comment:
```suggestion
            if (isAssignableToUnionType(source, target, unresolvedTypes)) {
```
Shall we remove that variable ?

Reply:
I will remove

Reply:
Updated. 4c82e4f1dfb0dba993c5a76e28b7ee7497fdc204